templates/media/media-with-message.php

```diff
@@ -0,0 +1,161 @@
+			jQuery( '#send-notice' ).click( function () {
+				if ( jQuery( this ).is( ':checked' ) ) {
+					jQuery( '#rtm_show_upload_ui' ).slideUp();
+				} else {
+					jQuery( '#rtm_show_upload_ui' ).slideDown();
+				}
+			} );
+		} );
+		jQuery( document ).ready( function() {
+			handler = function() {
+				var order = jQuery( '#messages_order' ).val() || 'ASC',
+				offset = jQuery( '#message-recipients' ).offset();
+				var button = jQuery( 'input#send_reply_button' );
+				jQuery(button).addClass( 'loading' );
+				jQuery.post( ajaxurl, {
+					action                  : 'messages_send_reply',
+					'cookie'                : bp_get_cookies(),
+					'_wpnonce'              : jQuery( 'input#send_message_nonce' ).val(),
+					'content'               : jQuery( '#message_content' ).val(),
+					'send_to'               : jQuery( 'input#send_to' ).val(),
+					'subject'               : jQuery( 'input#subject' ).val(),
+					'thread_id'             : jQuery( 'input#thread_id' ).val(),
+					'rtm_bpm_uploaded_media': jQuery( 'input#rtm_bpm_uploaded_media' ).val()
+				},
+				function( response )
+				{
+					if ( response[0] + response[1] == "-1" ) {
+						jQuery( 'form#send-reply' ).prepend( response.substr( 2, response.length ) );
+					} else {
+						jQuery( 'form#send-reply div#message' ).remove();
+						jQuery( '#message_content' ).val( '' );
+						jQuery( '#rtm_bpm_uploaded_media' ).removeAttr( 'value' );
+						if ( 'ASC' == order ) {
+							jQuery( 'form#send-reply' ).before( response );
+						} else {
+							jQuery( '#message-recipients' ).after( response );
+							jQuery(window).scrollTop( offset.top );
+						}
+					jQuery( '.new-message' ).hide().slideDown( 200, function() {
+						jQuery( '.new-message' ).removeClass( 'new-message' );
+					});
+					}
+					jQuery( button ).removeClass( 'loading' );
+				});
+				return false;
+			};
+			$( 'input#send_reply_button' ).unbind( 'click' ).bind( 'click', handler );
+		});
+
+	</script>
+	<span class="primary rtm-media-msg-upload-button rtmedia-upload-media-link" 
+		id="rtm_show_upload_ui" title="Upload Media">
+		<i class="dashicons dashicons-upload rtmicon"></i>
+			<?php echo esc_html__( 'Upload Media File', 'buddypress-media' ); ?>
+	</span>
+	<div id="rtm-media-gallery-uploader" class="rtm-media-gallery-uploader">
+		<?php
+		rtmedia_uploader(
+			[
+				'is_up_shortcode' => false,
+				'allow_anonymous' => true,
+				'privacy_enabled' => false,
+			]
+		);
+			?>
+	</div>
+	<input type="hidden" id="rtm_bpm_uploaded_media" name="rtm_bpm_uploaded_media" />
+	<?php
+}
+
+/**
+ * Add additional parameter media ID in BuddyPress message SEND MESSAGE process.
+ * And Insert data into MEDIA_META table.
+ *
+ * @param object $message Getting parameter response when sending message.
+ **/
+function rtm_add_message_media_params( $message ) {
+	$insert_media_object  = new RTDBModel( 'rtm_media_meta' );
+	$message->media_array = filter_input( INPUT_POST, 'rtm_bpm_uploaded_media' );
```

Comment:
You also need to pass `Sanitize filters` based on data type as third argument. or you can use `sanitize_text_field ()`